processor/src/test/kotlin/com/tobrun/datacompat/DataCompatProcessorTest.kt

```diff
@@ -27,8 +28,11 @@ interface EmptyInterface2
 @Deprecated
 @DataCompat
 private data class PersonData(
+    @Default("\"John\"")
```

Comment:
Why do we need to escape the `"` feels a bit wonky

Reply:
My guess is because there's no easy way to differentiate:
```
val name: String = "John"
```
from:
```
val name: String = "13"
```

Both are strings and if we allow `@Default("John")` then it's not clear what `@Default("13")` type is...

Actually, now that I write this I realize that we know the type of the `name` property so probably we could do a transformation on `defaultValue?.value` depending on the property type?

So for example, for:
```Kotlin
val numericId: Int = 13
val name: String = 13
```
Both could be:
```Kotlin
@Default("13")
val numericId: Int = 13
@Default("13")
val name: String = 13
```

To achieve above you can use this code in `DataCompatProcessor` line 225:
```Kotlin
                val stringClassName = String::class.asClassName()
                val nullableStringClassName = stringClassName.copy(true)
                val defaultValueString =
                    if (defaultValue != null && defaultValue.value == "null") {
                        "null"
                    } else if (defaultValue != null && (property.value == stringClassName || property.value == nullableStringClassName)) {
                        "\"${(defaultValue.value as? String?).toString()}\""
                    } else {
                        (defaultValue?.value as? String?) ?: "null"
                    }
```
and then use `defaultValueString` in line 230.
(You can probably make above code nicer 😄 )

Reply:
The code above won't work for typelias:
```Kotlin
typealias MyString = String

private data class PersonData(
    @Default("13")
    val name: MyString
)
```

But if you want to support typealias then you can use Kotlin poet tags to figure it out in the `DataCompatProcessor`:
```Kotlin
property.value.tag(TypeAliasTag::class)!!.abbreviatedType
```

Reply:
@jush thanks a lot for the code. However what about following use-cases:
```
class MyClass(
  val myStr = String.valueOf(56)
)
```
I guess it won't work.
As well as `""" """.trimIndent()` and perhaps some other corner cases...

IMO current way is not ideal but it covers all the use-cases as all the validation should be done by the user, we're not doing anything on top, simply putting it user's default value as is.

Reply:
You're right, my solution was assuming that for String the default value would be a constant or similar. 😿